### PR TITLE
Use name validations from our API instead of the k8s core

### PR DESCRIPTION
**Pull request by pmorie**
Pretty standard.

### CODE REVIEW

File: pkg/apis/servicecatalog/validation/binding.go [hunk 1]
```diff
@@ -23,13 +23,21 @@ import (
 	sc "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
 )
 
+// ValidateBindingName is the validation function for Binding names.
+var ValidateBindingName = apivalidation.NameIsDNSSubdomain
```
Comment (arschles) [inline pkg/apis/servicecatalog/validation/binding.go]: can you make this unexported?

File: pkg/apis/servicecatalog/validation/instance.go [hunk 2]
```diff
@@ -23,12 +23,20 @@ import (
 	sc "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
 )
 
+// ValidateInstanceName is the validation function for Instance names.
+var ValidateInstanceName = apivalidation.NameIsDNSSubdomain
```
Comment (arschles) [inline pkg/apis/servicecatalog/validation/instance.go]: same here RE making unexported

File: pkg/apis/servicecatalog/validation/serviceclass.go [hunk 3]
```diff
@@ -23,17 +23,25 @@ import (
 	sc "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
 )
 
+// ValidateServiceClassName is the validation function for ServiceClass names.
+var ValidateServiceClassName = apivalidation.NameIsDNSSubdomain
```
Comment (arschles) [inline pkg/apis/servicecatalog/validation/serviceclass.go]: same here RE making unexported

File: pkg/apis/servicecatalog/validation/binding.go (hunk 1 above)
Comment (duglin) [inline pkg/apis/servicecatalog/validation/binding.go]: EQ: Why the limitation on the chars being used? If there are kube rules that all names must conform to rfc2119 then ok, but I am kind of curious why there's a limitation when things can be escaped.

**Comment by duglin** [comment]
LGTM

File: pkg/apis/servicecatalog/validation/binding.go (hunk 1 above)
Comment (pmorie) [inline pkg/apis/servicecatalog/validation/binding.go]: I actually copied this from upstream - it seems unnecessarily restrictive - can I loosen in a followup?

Comment (duglin) [inline pkg/apis/servicecatalog/validation/binding.go]: @pmorie what about @arschles's comment?

File: pkg/apis/servicecatalog/validation/instance.go (hunk 2 above)
Comment (duglin) [inline pkg/apis/servicecatalog/validation/instance.go]: ping @pmorie

File: pkg/apis/servicecatalog/validation/serviceclass.go (hunk 3 above)
Comment (duglin) [inline pkg/apis/servicecatalog/validation/serviceclass.go]: @pmorie

**Comment by arschles** [comment]
@pmorie what's the status of this PR? If you still want to merge it, I'm ok with that as-is. I've approved my review and LGTM-ed as a result.

**Pull request closed without merging**